tests/integration-tests/tests/schedulers/test_custom_munge_key.py

```diff
@@ -0,0 +1,194 @@
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+import base64
+import os
+
+import boto3
+import pytest
+from assertpy import assert_that
+from remote_command_executor import RemoteCommandExecutor
+from retrying import retry
+from time_utils import minutes, seconds
+from utils import wait_for_computefleet_changed
+
+
+@pytest.mark.usefixtures("instance", "os")
+def test_custom_munge_key(
+    region,
+    pcluster_config_reader,
+    clusters_factory,
+    scheduler_commands_factory,
+    store_secret_in_secret_manager,
+    test_datadir,
+    s3_bucket_factory,
+):
+    """
+    Test custom munge key config, rotate, update, remove and roll back.
+
+    This test is focused on the scenario with LoginNodes.
+    Because this scenario covers all the logic covered by the test without login nodes。
+
+    Test phases summary:
+    1. Deployment verification: Confirm munge key is successfully shared across head, compute and login nodes, and jobs
+       can be submitted.
+    2. Rotation prep: Attempt munge key rotation without stopping compute and login nodes, expecting error messages.
+    3. Compute fleet shutdown: Prepare for munge key rotation scenario.
+    4. Key rotation without login node stop: Attempt rotation, checking for expected failure and error messages.
+    5. Login node stoppage and munge key rotation: Update cluster to stop login nodes, then execute munge key rotation.
+    6. Munge key removal: Update cluster to remove the custom munge key. Test Munge Key has been changed
+    7. Roll back with failure: Update cluster to add back the custom munge key. But let the cluster update fail after
+       the custom munge key has been added. Trigger cluster roll back. Test if munge key is fully functional.
+    """
+    encode_custom_munge_key = create_base64_encoded_key()
+    custom_munge_key_arn = store_secret_in_secret_manager(
+        region,
+        secret_string=encode_custom_munge_key,
+    )
+    cluster_config = pcluster_config_reader(custom_munge_key_arn=custom_munge_key_arn)
+    cluster = clusters_factory(cluster_config, upper_case_cluster_name=True)
+
+    # Test if the munge key was successfully fetched, decoded and shared in HeadNode and LoginNodes
+    remote_command_executor = RemoteCommandExecutor(cluster)
+    _test_custom_munge_key_fetch_and_decode(remote_command_executor, encode_custom_munge_key)
+    _test_munge_key_shared(remote_command_executor)
+
+    remote_command_executor_login = RemoteCommandExecutor(cluster, use_login_node=True)
+    _test_custom_munge_key_fetch_and_decode(remote_command_executor_login, encode_custom_munge_key)
+    remote_command_executor_login.close_connection()
```

Comment:
Please use this connection to execute a job from the login nodes too before closing it.

Reply:
Done!